js/coinsbit.js

```diff
@@ -0,0 +1,682 @@
+        return this.status;
+    }
+
+    async fetchTime (params = {}) {
+        return this.milliseconds ();
+    }
+
+    async fetchMarkets (params = {}) {
+        const [exchangeType, request] = this.checkExchangeType ('fetchMarkets', params); // Note: imho, I am against dividing methods according to types (like 'fetchSpotMarkets', 'fetchFuturesMarkets' and etc..) because there might be many types and exchanges, from time to time, add new types, thus it will need a never-ending 'new method' creations. I think that, independent to the exchange-type, everything should be done with one 'fetchXXX' functions - if there are multiple different types, they should be handled inside one 'fetchXXX' function (just by passing 'exchangeType' parameter). That has also one advantage - the most of the 'return' object structure can be same and only a few property-values might need to be happening inside if-else blocks, thus will save multiple 'fetchSpotXXX', 'fetchFuturesXXX' functions to be written over-and-over again with same structure.
+        let data = '';
+        if (exchangeType === 'spot') {
+            const response = await this.spotPublicGetMarkets (request);
+            //     {
+            //         "success": true,
+            //         "message": "",
+            //         "result": [
+            //             {
+            //                 name: "BTC_USDT",
+            //                 moneyPrec: "8",
+            //                 stock: "BTC",
+            //                 money: "USDT",
+            //                 stockPrec: "8",
+            //                 feePrec: "8",
+            //                 minAmount: "0.001"
+            //             },
+            //             ...
+            //          ],
+            //         "code":"200",
+            //
+            data = this.safeValue (response, 'result', []);
+        } else if (exchangeType === 'futures') {
+            // JUST FOR DEMONSTRATIONAL PURPOSES
+            // const response = await this.futuresPublicGetMarkets (params);
+            // i.e. : data = this.safeValue (response, 'result', []);
+        }
+        const result = [];
+        for (let i = 0; i < data.length; i++) {
+            const market = data[i];
+            const id = this.safeString (market, 'name');
+            // const [ baseId, quoteId ] = id.split ('_');  // We do have them in properties, so, I don't think this is needed ?
+            const baseId = this.safeString (market, 'stock');
+            const quoteId = this.safeString (market, 'money');
+            const base = this.safeCurrencyCode (baseId);
+            const quote = this.safeCurrencyCode (quoteId);
+            const symbol = base + '/' + quote;
+            const priceScale = this.safeInteger (market, 'moneyPrec'); // <<< TODO >>>  strings 1e-8
+            const quantityScale = this.safeInteger (market, 'stockPrec');
+            const pricePrecision = 1 / Math.pow (10, priceScale);
```

Comment:
We should probably use precisionMode = DECIMAL_PLACES in this case or use string-based math in all parsers to avoid floating-point math as much as we can:

- https://docs.ccxt.com/en/latest/manual.html#precision-and-limits
- https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#adding-numbers-and-concatenating-strings

Reply:
This is the only thing that remains to be done from review.

Reply:
Done, please check the PR and let me know if that's what you meant.